### DISCUSSION TITLE
Fitting podcasts into beets

### DISCUSSION THREAD

**Discussion by Serene-Arc** [beetbox/beets#4542] [Ideas]
I'm thinking of making a plugin to organise podcasts and podcast episodes. I'm wondering if anyone has any ideas about how to fit these into the system that beets uses. As I see it, there are a couple of different ways to do it:

- Treat each episode as a distinct album and use the path configuration to keep them in the same directory, with the artist being the name of the podcast
  - Seems like the best option, although it would split up the podcast in a manner that seems counterintuitive
- Import each episode as singletons
  - Has a fairly big drawback in that ordering would be impossible
  - Would make it easier to import new episodes as they are released
- Import each podcast as an album
  - would make adding new episodes extremely difficult
  - would make adding partial collections difficult

I'm leaning towards the first option, but I'm curious if there are any other thoughts! I've also got to work out how to make a plugin that provides an alternative media source but there's a few examples..

**Comment by sampsyo** [comment]
Wow; that's super interesting!!! Sounds like a fun project!

I confess I haven't thought about it terribly hard, but I think the second option could also be promising. That is, if every episode is a singleton, then you'd use _other_ metadata to associate episodes with the podcasts they're a part of. For example:

* You could use a custom `podcast` field to store the name of the podcast.
* And an `episode` (integer-typed) field to store the episode number.

Then, I suppose, you could list all the episodes of a podcast in order with a query like `beet ls podcast:'This American Life' episode+`. It would be nice if that `episode+` ordering constraint were somehow implicit, purely for convince, but that seems like another issue…

**Reply by Serene-Arc** [reply]
I was thinking that the podcast name would be the artist name, since a lot of podcasts don't have the names of the individual people involved. Ordering the podcasts is obviously very important. Many podcasts also have seasons or groupings that could act as an album, though distinguishing them algorithmically would be a significant challenge.

I suppose the paradigm for the library would be similar to that of the beets-audible plugin: a separate library that contains no music, only podcasts. That way there are minimal interactions the plugin can be a lot more adventurous with the manipulation of metadata.

**Reply by sampsyo** [reply]
Sure—reusing the `artist` or `albumartist` field would work as well! I guess with that in mind, I don't quite see the advantage of using an album… it seems like the metadata would be enough to group things together (and order episodes), with or without actually using an album entry in the database. (But perhaps I'm missing something!)

**Reply by Serene-Arc** [reply]
Fair, albums do seem unnecessary. I'll see what I come up with!
